src/cmd/factory/impl/common.go

```diff
@@ -179,3 +199,547 @@ func buildCollections(
 
 	return collections, nil
 }
+
+type permData struct {
```

Comment:
These are all copied from `graph_connector_onedrive_test.go`, right?  Instead of duplication, I'd rather see those moved into something like `onedrive/stub/collections.go` (or testdata, if not stub) and exported to both the test and to this file.  Normally we wouldn't want to export testdata packages to non-test code, but since this is an isolated helper, not part of the corso binary, it'll be safe to do.

By sharing this code we have much less duplication to track and maintain, and won't have this factory getting out of sync across version changes.  I bet that consolidation alone would bring the changes in this file from ~700 lines down to one or two hundred.

For the sake of time, we can move forward with the duplication as it exists here, but the refactor should come quickly after that.

Reply:
I will get the PR with this refactoring ready soon.